src/Scripting/Core/ScriptState.cs

```diff
@@ -59,13 +59,13 @@ public ImmutableArray<ScriptVariable> Variables
         /// <summary>
         /// Returns a script variable of the specified name. 
         /// </summary> 
+        /// <returns>True if variable of the specified <paramref name="name"/> is defined in the script, false otherwise.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
         /// <remarks>
         /// If multiple script variables are defined in the script (in distinct submissions) returns the last one.
-        /// Namve lookup is case sensitive in C# scripts and case insensitive in VB scripts.
+        /// Name lookup is case sensitive in C# scripts and case-insensitive in VB scripts.
         /// </remarks>
-        /// <returns><see cref="ScriptVariable"/> or null, if no variable of the specified <paramref name="name"/> is defined in the script.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
-        public ScriptVariable GetVariable(string name)
+        public bool TryLookupVariable(string name, out ScriptVariable variable)
```

Comment:
Why not TryGetVariable? The API actually gets the variable if it exists. I think the feedback in the meeting was to make it a "Try" method because the variable may not exist. I haven't see the word "Lookup" in any BCL APIs. It is a very compiler design concept that you lookup and find the appropriate variable when there can be more than one but we probably should not bleed that into the API name. For user he should only see one and the correct one.